hayeon/programmers-1/바탕화면-정리.js

```diff
@@ -0,0 +1,35 @@
+function solution(wallpaper) {
+  const row = [];
+  const column = [];
+  let answer = [];
+
+  const width = wallpaper[0].length;
+  const height = wallpaper.length;
+
+  for (let i = 0; i < height; i++) {
+    for (let j = 0; j < width; j++) {
+      if (wallpaper[i][j] === "#") {
+        row.push(j);
+        column.push(i);
+      }
+    }
+  }
+
+  row.sort((a, b) => a - b);
+  column.sort((a, b) => a - b);
+
+  answer = [column[0], row[0], Math.max(...column) + 1, Math.max(...row) + 1];
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(wallpaper) {
  const row = [];
  const column = [];

  const width = wallpaper[0].length;
  const height = wallpaper.length;

  for (let i = 0; i < height; i++) {
    for (let j = 0; j < width; j++) {
      if (wallpaper[i][j] === "#") {
        row.push(j);
        column.push(i);
      }
    }
  }

  row.sort((a, b) => a - b);
  column.sort((a, b) => a - b);

  return [column[0], row[0], Math.max(...column) + 1, Math.max(...row) + 1];
}
```
- 풀이 잘봤습니다~
- answer부분을 그냥 return하면 answer 변수는 불 필요한 변수가되서 제거할 수 있습니다.
- 아래와 같은 풀이로 sort와 answer 부분 등 가독성있게 개선할 수 있습니다.
```js
function solution(wallpaper) {
  const result = [50, 50, 0, 0];
  const height = wallpaper.length;
  const width = wallpaper[0].length;

  for (let i = 0; i < height; i++) {
    for (let j = 0; j < width; j++) {
      if (wallpaper[i][j] === "#") {
        result[0] = Math.min(result[0], i);
        result[1] = Math.min(result[1], j);
        result[2] = Math.max(result[2], i + 1);
        result[3] = Math.max(result[3], j + 1);
      }
    }
  }

  return result;
}
```

Reply:
- 우와 이렇게 풀 수도 있군요!!!!!! 훨씬 깔끔해보이네요.. 참고해보겠습니다!!👍🏻